programs/uxd/src/instructions/redeem.rs

```diff
@@ -160,16 +163,66 @@ struct DepositoryInfoForRedeem<'info> {
 
 pub(crate) fn handler(ctx: Context<Redeem>, redeemable_amount: u64) -> Result<()> {
     // Gather all the onchain states we need (caps, weights and supplies)
-    let controller = ctx.accounts.controller.load()?;
+    let mut controller = ctx.accounts.controller.load_mut()?;
     let identity_depository = ctx.accounts.identity_depository.load()?;
     let mercurial_vault_depository = ctx.accounts.mercurial_vault_depository.load()?;
     let credix_lp_depository = ctx.accounts.credix_lp_depository.load()?;
+    let now_timestamp = Clock::get()?.unix_timestamp;
+
+    // Limit per day (NO MERGE, this include both options!! we only need one!!)
+    let limit_outflow_amount_per_day = if true {
+        controller.limit_outflow_amount_per_day.into()
+    } else {
+        controller
+            .redeemable_circulating_supply
+            .checked_mul(1000 /* controller.limit_outflow_bps_per_day */)
+            .ok_or(UxdError::MathError)?
+            .checked_div(BPS_UNIT_CONVERSION.into())
+            .ok_or(UxdError::MathError)?
+    };
+
+    // How much was unlocked by waiting since last redeem
+    let unlocked_outflow_amount = checked_convert_u128_to_u64({
+        u128::try_from(
+            now_timestamp
+                .checked_sub(controller.last_redeem_timestamp)
+                .ok_or(UxdError::MathError)?,
+        )
+        .ok()
+        .ok_or(UxdError::MathError)?
```

Comment:
Could we base this PR off the Math PR?

Reply:
I think all new code should use the new maths to not create more refactor for later, if you'r okay with the math PR

Reply:
Done 👌 

Merged math PR into this one 👌

Reply:
Also used the new functions 🥳